streams/src/main/java/org/apache/kafka/streams/kstream/Materialized.java

```diff
@@ -53,6 +53,30 @@
  *   "topicName",
  *   Materialized.as("queryable-store-name"));
  * }</pre>
+ * <p>
+ * <b>Correct Usage When Providing Serde:</b>
+ * <p>
+ * To configure both the name of the store and the {@link Serde} for the key and value,
+ * you should use the following pattern:
+ * <pre>{@code
+ * Materialized.<KeyType, ValueType, StateStore>as("MyStoreName")
+ *     .withKeySerde(keySerde)
+ *     .withValueSerde(valueSerde);
+ * }</pre>
+ * This ensures that the store name is retained while configuring the key and value serde.
+ * <p>
+ * <b>Warning:</b> If you use the {@link Materialized#with(Serde, Serde)} method after calling
+ * {@link Materialized#as(String)}, the instance created by {@link Materialized#as(String)} will be replaced by a new
+ * {@link Materialized} instance, and any configuration set on the first instance (e.g., store name, logging settings)
+ * will be lost.
+ * <p>
+ * For example, the following code is incorrect because it discards the configuration of the store name:
```

Comment:
```suggestion
 * For example, the following code is incorrect because it discards the configuration of the store name (calling {@code static}) methods on an object/instance is an anti-pattern):
```